### ISSUE TITLE
Blokads is blocking downloads that use the Android Download Manager service

### ISSUE THREAD

**Issue by thanuj10** [blokadaorg/blokada#324]
This is a very critical bug that needs to be fixed. Any app or that uses the Android Downloads manager is blocked by blokada and i do not know why it does it. Download Manager is already added in the app whitelist. This mostly happens with browsers that use WebView and depend on androids download manager for downloads.
Chrome and Firefox and browsers that use their internal download manager is not affected.

### Steps to reproduce
1. Open Via or similar browsers.
2. Download something.
3. Download does not start.

### Expected behaviour
Downloads must start and finish.

### Actual behaviour
Downloads do not start and nothing is added to the download queue. Once Blokada is stopped and download reatarted, Download starts and resumes normally.

### Configuration
1. Blokada version - 3.6 (Latest Release)

2. Device model - Nokia 6.1 Plus

3. Android version / ROM - Android 9.0/Pie

### Questions
1. How long have you been using Blokada? 
About a year

2. How long time ago has the problem started to happen? 
Almost always, never reported.

3.  How often does this problem happen to you?
Everytime i download something.

**Comment by rpeter85** [comment]
Hi,

Thanks for reporting it. Are you use the download doesn't start? If you want, doesn't your phone report that the download finished? The package com.android.providers.downloads.ui isn't whitelisted, this is the user interface of the com.android.providers.downloads package. Can you try to add the earlier to the whitelist and test the behaviour again?

The criticality of a bug depends on a lot of factors. This is not critical and not yet identified as a bug.

**Comment by thanuj10** [comment]
Yes i'm very sure the download does not start.
For more clear instructions on repo the bug -
1. Download Via or Firefox Lite(i tested both, Via is bit more easier to work with)
2. Download any file, I tried downloading Bromite.apk and Firefox Fenix from Github and Taskcluster respectively.
3. Starting download shows a prompt on the screen indicating the download has started, but there is no notification or the download isnt added to the download list in downlaods app.
4. Checking the browsers downloads page i get this -
Firefox Lite - Nothing is shown
Via - A download is shown, but the filesize is shown incorrect (0bytes of -1.0bytes).
5. Turning of blokada makes the browser(s) resume the download.

Ill try whitelisting the app you mentioned above, Thanks.

**Comment by animeking9901** [comment]
Let us know as if its that simple,the UI needing a whitelisting,that could
be defaulted if needed. Also can confirm at least with the LineageOS
default,this used to baffle me that downloads never worked but I figured
try later. Thought it was just me!

**Comment by rpeter85** [comment]
@thanuj10 could you elaborate on the "The app exhibits the same behaviour."?

**Comment by thanuj10** [comment]
@rpeter85 i meant that the app still blocks downloads after whitelisting com.android.providers.downloads.ui

**Comment by larabih** [comment]
I confirm the same incident.
I use Firefox Focus and when I download a file, the download start icon appears briefly and disappears immediately.
I disable Blockada and re-download the file and then everything works fine.
the same behavior occurs when I use webview applications such as Frost for facebook 

**Configuration**
-   Blokada version - 3.6 (Latest Release)
-   Device model - Oneplus One
-   Android version / ROM -  /e/ e.fondation system version 0.5 based on Android nougat 7.1.

**Questions**
    How long have you been using Blokada?
   since the version 2.x.x

    How long time ago has the problem started to happen?
    when i was on LineageOS 15 and now on /e/ system

    How often does this problem happen to you?
    Everytime i download something

**Comment by rpeter85** [comment]
Can you tell me which browser do you face the issue with? Firefox Focus, is there another one?

**Comment by thanuj10** [comment]
Im my case all webview based apps and browsers have this problem. Via, Firefox Lite, Firefox Focus.

**Comment by larabih** [comment]
@rpeter85  : 
**Browsers:** 
- Firefox focus : download don't work
- Firefox : download  works fine 
- Bromite (based on chromium): download  works fine 

**Apps**
- Frost for facebook: an open source wrapper of the facebook mobile version. It have option  to download fb videos  but it don't work when Blokada is running.

**Comment by thanuj10** [comment]
As i said before, Browsers with an internal download manager work fine. But Webview based apps do not download stuff.

**Comment by matteocoder** [comment]
Can the issue be caused by this [bug](https://issuetracker.google.com/issues/111574523)? 
@thanuj10 Do you have any devices with Android 6 or earlier to see if you can reproduce the issue?

**Comment by thanuj10** [comment]
Yeah i do actually. Ill try reproducing the bug devices with android 6 or earlier.

Side note - This bug exists on Android 7.1.2 also

**Comment by racuna** [comment]
Same here, MIUI 10, Android 8.1

**Comment by thanuj10** [comment]
@rpeter95 Seems like other adblockers that use similar mechanism like Blokada allow downloads through android's download manager when whitelisted. 

Source - Adgaurd and DNS66 both allow downloads when android's download manager is whitelisted. 

Seems like a problem with the default whitelist, maybe in the next update let blokada whitelist all system apps by default?

**Comment by rpeter85** [comment]
In v4, system apps can be whitelisted as well :)

**Comment by thanuj10** [comment]
Ah, then the problem might be sorted. Excited for v4's release!

**Comment by Buntbart** [comment]
Same here: LOS 16 without GApps, 
Browsers: Lightning, Firefox Klar, Privacy Browser, FOSS Browser. All fail to start downloads when Blokada is active

**Comment by larabih** [comment]
> In v4, system apps can be whitelisted as well :)

If apps can be whitelisted, blokada will not block ads domains from this apps or I misunderstood ?

**Comment by thanuj10** [comment]
@larabih yes whitelisting will allow ads to show. But generally speaking, system apps do not show ads.

**Comment by thanuj10** [comment]
Can i ask about the progress on this issue? Im forced to use other adblocking apps as this blocks download manager and i use it alot..

**Comment by larabih** [comment]
@thanuj10 
This problem was confirmed on the ROM LineageOS. Personally I use firefox to open download links instead of Firefox Focus which I usually use.

**Comment by Herbstfreud** [comment]
I opened this as Issue #209 already. Seems to be a kind of restriction in Android itself.

A browser which can handle downloads with Blokada enabled is Fennec, also available at F-Droid.

**Comment by iriseecz** [comment]
Same problem here (Blokada 3.7 installed from F-Droid, on Android 9 / Nokia 6.1 / Android One, Firefox Focus 8.0.11 installed from Google Play) - cannot download files, but without Blokada it works (tested on Nokia 5 with same Firefox Focus).

**Comment by rpeter85** [comment]
Can you let me know what exactly should be whitelisted to be able to download things? I want to add that to default whitelist.

**Comment by Herbstfreud** [comment]
There is something that can be whitelisted for that? This would be really interesting - I haven't seen anything for this.

Or is there some type of alternative download manager or alternative webview engine that can be installed?

**Comment by soredake** [comment]
Any progress on this?

**Comment by thanuj10** [comment]
@rpeter85 I've seen some other VPN Based adblockers whitelist the Download Manager app by default.

**Comment by SeriousHoax** [comment]
I face the same problem too. Happens on Firefox Focus and a third party twitter app that I use named, Flamingo for twitter. Download manager is whitelisted but still this happens. I'm forced to use another adblocker at the moment. Seen some version 4 screenshots and seems like finally it's gonna have everything we want feature wise. Hopefully a solution for this problem would be found before the release of version 4.

**Comment by thanuj10** [comment]
Problem still not fixed in V4.

**Comment by SeriousHoax** [comment]
Well, version 4 got released today! Didn't notice. This problem needs to be fixed soon.

**Comment by rpeter85** [comment]
Do you guys know what is the cause? As it currently seems a system app is blocked, and they can be whitelisted in v4. Did you try that?

**Comment by thanuj10** [comment]
@rpeter85 the download manager service is already blacklisted by default. Also i added the download provider apk(The downloads app). No changes even after that.

**Comment by kar** [comment]
Is download manager working for you under Blokada VPN? Anyone tried?

**Comment by kar** [comment]
Also try enabling downloads on any network (not only wifi) in google play

**Comment by jleuwen** [comment]
I have checked downloading when not on WIFI and this works over here. However Blokada is blocking the download (manager) when I am downloading while I am on WIFI. When I will disable Blokada, downloads can be done again on WIFI.  I am running Blokada 4.3.5 on a Xiaomi Mi9T (Global standard Rom, MIUI 10.3.11).

**Comment by kar** [comment]
Anyone experiencing this problem, if you're on Telegram please reach out to me in priv.

**Comment by Buntbart** [comment]
Yes, same here. No Google Play Services installed, LineageOS 16 on klte. Downloads in Lightning-Browser do not work when connected via WIFI.

**Comment by PrintableCharacter** [comment]
@Buntbart could you DM Karol on Telegram? 😉

**Comment by Buntbart** [comment]
I don't see why that should be necessary since he is able to read and write here and I don't use Telegram.

**Comment by rpeter85** [comment]
To have a live discussion and not only messages in every half a day - 2 days. Easier to help that way.

**Comment by afonari** [comment]
This also affects downloading Apps on /e/: https://community.e.foundation/t/e-app-store-is-live/3803/10

**Comment by jleuwen** [comment]
> 
> 
> I have checked downloading when not on WIFI and this works over here. However Blokada is blocking the download (manager) when I am downloading while I am on WIFI. When I will disable Blokada, downloads can be done again on WIFI. I am running Blokada 4.3.5 on a Xiaomi Mi9T (Global standard Rom, MIUI 10.3.11).

I can confirm that the latest version solves the issue. However, it was not working before deleting all the (system) apps from the exclusions and after that including them again. After including the download system app (2 times), all files will download also on WIFI. Good job.

**Comment by Buntbart** [comment]
No change in LineageOS 16 with Blokada 4.4.3 from F-Droid. Downloads are blocked when on Wifi. I excluded/included Download-Manager as described in the last post.

**Comment by eXPerience83** [comment]
Can't download updates or apps from Xiaomi Store "GetApps" if I didn't whitelist it.
And I can't see the blocked petition on the blockada log. I see that Xiaomi enables by default the "Secure DNS" and maybe this is the problem.

**Comment by bigtreiber** [comment]
I too have the same issue with Blokada 4.8.5 on Android 11 (OxygenOS). I tried excluding / including the "Download-Manager" on the whitelist but still can't download anything that is using webview :-(

**Comment by SeriousHoax** [comment]
> I too have the same issue with Blokada 4.8.5 on Android 11 (OxygenOS). I tried excluding / including the "Download-Manager" on the whitelist but still can't download anything that is using webview :-(

Switch to NextDNS or Adguard DNS like I did.

**Comment by tenzap** [comment]
I'm facing that issue too on Xiaomi redmi note 9s, Android 11, blokada 5. I try to download fdroid app from fdroid website with duckduckgo app which then uses the built-in download manager. But download doesn't start.

Changing dns to adguard or using network DNS doesn't fix. Only way to get things downloaded is by disabling blokada.

**Comment by tenzap** [comment]
Since I don't know if this issue is followed, I just do a small @kar who seem to be in blokada team.

Btw what kind of additional input do you need for this issue?

As a addition to me previous comment, whitelisting these doesn't fix either:
com.android.providers.downloads
com.android.providers.downloads.ui
